staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -1503,9 +1503,7 @@ type DeveloperConfiguration struct {
 // PermittedHostDevices holds inforamtion about devices allowed for passthrough
 // +k8s:openapi-gen=true
 type PermittedHostDevices struct {
-	// +listType=set
-	PciHostDevices []PciHostDevice `json:"pciHostDevices,omitempty"`
-	// +listType=set
+	PciHostDevices  []PciHostDevice      `json:"pciHostDevices,omitempty"`
 	MediatedDevices []MediatedHostDevice `json:"mediatedDevices,omitempty"`
```

Comment:
if `set` is the wrong listType here, what should we be marking it as? `listType=atomic` ?

Reply:
Yes or `listType=map`. `atomic` would always replace whole slice and with `map` you provide key based on which the update will be performed.

Reply:
should this be updated? instead of adding it as an "API rule violation:" to the api-rules-validations-* lists?

Reply:
It is not necessary for this PR and unless we decide to support SSA. This can be 'fixed' anytime in future.

Reply:
If we're going to change it in this PR, we can update it in this PR instead of adding a new violation. listType=atomic should be fine.